Source/Atomic/Core/Context.h

```diff
@@ -228,6 +228,27 @@ class ATOMIC_API Context : public RefCounted
     void AddGlobalEventListener(GlobalEventListener* listener) { globalEventListeners_.Push(listener); }
     void RemoveGlobalEventListener(GlobalEventListener* listener) { globalEventListeners_.Erase(globalEventListeners_.Find(listener)); }
 
+    WeakPtr<Engine> engine_;

```

Comment:
C#, JS/TS properties are automatically generated for ```Getxxx``` methods so whether the WeakPtr's are private or public the methods should exist, Vector3 is a primitive and you definitely want direct access to ```x_, y_, z_``

My spidey sense is telling me private WeakPtr and public accessors, and not both... 🕷️